unit_testing/DC/DC_tests.cpp

```diff
@@ -0,0 +1,506 @@
+    ** printing anything */
+    DCBoardLoop(bootMsg);
+    writeDcMessage("Status\n");
+    
+
+    EXPECT_FLUSH_USB(ElementsAre(
+        "\r", 
+        "Boot Unit Test\r", 
+        "Board Status:\r", 
+        "The board is operating normally.\r", 
+        "Port 0: On: 0, PWM percent: 0\r", 
+        "Port 1: On: 0, PWM percent: 0\r", 
+        "Port 2: On: 0, PWM percent: 0\r", 
+        "Port 3: On: 0, PWM percent: 0\r", 
+        "Port 4: On: 0, PWM percent: 0\r", 
+        "Port 5: On: 0, PWM percent: 0\r", 
+        "\r", 
+        "End of board status. \r"
+    ));
+    
+    writeDcMessage("p2 on\n");
+    writeDcMessage("p4 on\n");
+    writeDcMessage("p6 on\n");
+    writeDcMessage("Status\n");
+    
+    EXPECT_FLUSH_USB(ElementsAre( 
+        "\r", 
+        "Board Status:\r", 
+        "The board is operating normally.\r", 
+        "Port 0: On: 0, PWM percent: 0\r",
+        "Port 1: On: 1, PWM percent: 999\r",
+        "Port 2: On: 0, PWM percent: 0\r",
+        "Port 3: On: 1, PWM percent: 999\r",
+        "Port 4: On: 0, PWM percent: 0\r",
+        "Port 5: On: 1, PWM percent: 999\r",
+        "\r", 
+        "End of board status. \r"
+    ));
+}
+
+TEST_F(DCBoard, status24v) 
+{
+    dcSetup();
+    stmSetGpio(sense24v, false);
+
+    /* Note: usb RX buffer is flushed during the first loop, so a single loop must be done before
+    ** printing anything */
+    goToTick(100);
+    writeDcMessage("Status\n");
+    
+
+    EXPECT_FLUSH_USB(ElementsAre(
+        "\r", 
+        "Boot Unit Test\r", 
+        "0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0xa0000000\r",
+        "Board Status:\r", 
+        "Under voltage. The board operates at too low voltage of 0.00V. Check power supply.\r",
+        "Port 0: On: 0, PWM percent: 0\r", 
+        "Port 1: On: 0, PWM percent: 0\r", 
+        "Port 2: On: 0, PWM percent: 0\r", 
+        "Port 3: On: 0, PWM percent: 0\r", 
+        "Port 4: On: 0, PWM percent: 0\r", 
+        "Port 5: On: 0, PWM percent: 0\r", 
+        "\r", 
+        "End of board status. \r"
+    ));
+}
+
+/* Grey box - uses getTimerCCR() */
+TEST_F(DCBoard, portsNoTimeout) 
+{
+    dcSetup();
+
+    goToTick(1);
+    
+    char cmd[100] = {0};
+    for(int i = 0; i < 8; i++) {
+        sprintf(cmd, "p%u on\n", i);
+        writeDcMessage(cmd);
+
+        if(i == 0 || i == 7) {
+            for(int j = 0; j < ACTUATIONPORTS; j++) {
+                ASSERT_EQ(*getTimerCCR(j), 0) << "j = " << j;
+            }
+            sprintf(cmd, "MISREAD: Invalid Pin: %d", i);
+            EXPECT_FLUSH_USB(Contains(cmd));
+        }
+        else {
+            for(int j = 1; j <= ACTUATIONPORTS; j++) {
+                if(j != i) {
+                    ASSERT_EQ(*getTimerCCR(j-1), 0) << "j = " << j;
+                }
+                else {
+                    ASSERT_EQ(*getTimerCCR(j-1), 999) << "j = " << j;
+                }
+            }
+        }
+
+        writeDcMessage("all off\n");
```

Comment:
Would it make sense to see that the port that had just been set now has a CCR of 0 again?

Reply:
Wait it is already checked in the next loop iteration of course.